ctlearn/build_irf.py

```diff
@@ -375,37 +389,181 @@ def main():
-    for tab in (gammas, background):
-        tab["selected_gh"] = evaluate_binned_cut(
-            tab["gh_score"], tab["reco_energy"], gh_cuts, operator.ge
+    
+ 
+    if args.energy_dependent_gh_efficiency:
+        
+        log.info(f"Performing a energy dependant G/H cut using gamma efficiency of {args.energy_dependent_gh_efficiency}")
+        gh_cuts_coarse = calculate_percentile_cut(
+            gammas["gh_score"],
+            gammas["reco_energy"],
+            bins=sensitivity_bins,
+            min_value=0.05,
+            max_value=0.95,
+            fill_value=gammas["gh_score"].max(),
+            percentile=100 * (1 - args.energy_dependent_gh_efficiency),
+            smoothing=None,
+            min_events=100,
+        )   
+
+        gh_cuts = table.QTable(
+            {
+                "low": reco_energy_bins[:-1],
+                "high": reco_energy_bins[1:],
+                "center": gh_center,
+                "cut": np.interp(
+                    np.log10(gh_center / u.TeV),
+                    np.log10(inter_center / u.TeV),
+                    gh_cuts_coarse["cut"],
+                    ),
+            }
         )
+        
+        for tab in (gammas, background):
+            tab["selected_gh"] = evaluate_binned_cut(
+                tab["gh_score"], tab["reco_energy"], gh_cuts, operator.ge
+                )
+
+        gammas["selected_theta"] = evaluate_binned_cut(
+            gammas["theta"], gammas["reco_energy"], theta_cuts, operator.le
+            )
+        gammas["selected"] = gammas["selected_theta"] & gammas["selected_gh"]
+        
+        sensitivities = []
+        
+        gammas_hist = create_histogram_table(
+            gammas[gammas["selected"]], sensitivity_bins, "reco_energy"
+            )
+        
+        background_hist = estimate_background(
+            events=background[background["selected_gh"]],
+            reco_energy_bins=sensitivity_bins,
+            theta_cuts=theta_cuts,
+            alpha=ALPHA,
+            fov_offset_min=FOV_OFFSET_MIN,
+            fov_offset_max=FOV_OFFSET_MAX
+            )
 
-    gammas["selected_theta"] = evaluate_binned_cut(
+        sensitivity_coarse = calculate_sensitivity(
+            gammas_hist, background_hist, alpha=ALPHA
+            )
+        sensitivities.append(sensitivity_coarse)
+        
+        sensitivity = sensitivities[0].copy()
+        for bin_id in range(len(sensitivity_bins) - 1):
+            sensitivities_bin = [s["relative_sensitivity"][bin_id] for s in sensitivities]
+            
+        if not np.all(np.isnan(sensitivities_bin)):
+            # nanargmin won't return the index of nan entries
+            best = np.nanargmin(sensitivities_bin)
+        else:
+            # if all are invalid, just use the first one
+            best = 0
+                    
+        sensitivity[bin_id] = sensitivities[best][bin_id]
+         
+    
+        
+    elif args.global_gh_cut:
```

Comment:
What happens if I select both, the `energy_dependent_gh_efficiency` and `global_gh_cut`? I think that `energy_dependent_gh_efficiency` should have higher priority so that the argument for `global_gh_cut` should be ignored in this case.
I guess you need here (please check syntax):
`if args.global_gh_cut and not bool(args.energy_dependent_gh_efficiency):`